botocore/config.py

```diff
@@ -235,6 +235,37 @@ class Config:
         specified service will be ignored.
 
         Defaults to None.
+
+    :type request_checksum_calculation: str
+    :param request_checksum_calculation: Determines when a checksum will be
+        calculated for request payloads. Valid values are:
+
+        * ``when_supported`` -- When set, a checksum will be calculated for
+            all request payloads of operations modeled with the ``httpChecksum``
+            trait where ``requestChecksumRequired`` is ``true`` and/or a
+            ``requestAlgorithmMember`` is modeled.
+
+        * ``when_required`` -- When set, a checksum will only be calculated
+            for request payloads of operations modeled with the ``httpChecksum``
+            trait where ``requestChecksumRequired`` is ``true`` or where a
+            ``requestAlgorithmMember`` is modeled and supplied.
+
+        Defaults to None.
```

Comment:
We say `Defaults to None.` for a few options in this file which is correct, however, during evaluation there may be an actual default value used. For both checksum config options, we default to `when_supported`. The reason for defaulting to None here and not `when_supported` is to prevent overwriting any values a user might configure at the session level, config file, or environment variable.

I wanted to point this out because it seems confusing from a user perspective when they read `Defaults to None.` and experience different behavior. How do we feel about updating our user documentation to mention this nuance.